Refactored the Authentication logic to use Flask-login instead of repoze.who

kovalch opened the pull request:
According to https://docs.ckan.org/en/2.11/changelog.html#id307

kovalch committed:
feat: Upgrade to py3 using 2to3

kovalch committed:
fix: Remove python-future that was used to make the scripts py2 compatible

kovalch committed:
fix: Update init file

kovalch committed:
fix: Install the missing webob module

kovalch committed:
fix: Remodjust import to work with either CKAN 2.8 or 2.9+

kovalch committed:
feat: Refactored the Authentication logic to use Flask-login instead of repoze.who

bellisk commented on requirements.txt:
```diff
@@ -1 +1,2 @@
 future==1.0.0
+webob>=1.8.0
```
We are not using webob any more, so this can go.

bellisk commented on ckanext/password_policy/views.py:
```diff
@@ -129,232 +111,67 @@ def _get_form_password(self):
+            flash(helper.lockout_message(), 'error')
+            return redirect(url_for('custom_user.locked'))
 
+        user_obj = helper.authenticate_user(username, password)
```
This function doesn't seem to exist. Where are you getting it from?

bellisk commented on ckanext/password_policy/views.py:
```diff
@@ -129,232 +111,67 @@ def _get_form_password(self):
-    alert = helper.lockout_message()
+        if helper.user_locked_out(username):
+            flash(helper.lockout_message(), 'error')
+            return redirect(url_for('custom_user.locked'))
```
We need to log that the user has tried to log in again despite being locked out, and also increment the login attempt count. This was done in the old code and should be in the new code as well.

There are several places like this where log statements used to exist and don't any more. Please re-add them.

stefina committed:
feat: copy template from CKAN 2.11.3 to override it

stefina committed:
style: fix indentation to protect me from eyesore

stefina committed:
feat: Enable sysadmins to reset locked-out users

Re-apply the commit with the actual change of the template to enable sysadmins to reset locked-out users.

stefina committed:
feat: authenticate user through CKANs authenticator

stefina committed:
fix: forward user to dashboard when logging in

stefina committed:
chore: remove webob as we do not use it
